### PR TITLE
Convert To Qbeast

### PR DESCRIPTION
**Pull request by osopardo1**
## Description

Adds feature #102 

The idea is to have a way of converting an already existing table (either written in `Parquet` or `Delta`) into the `Qbeast` Format. 

To do so, I added two commands called `ConvertToQbeast` and `PartiallyConvertToQbeast` for experimenting the two approaches proposed in the issue.

## Type of change

It is a **New Feature**  (non-breaking change, which adds new functionality).

## Checklist:
- [x] It's a WIP
- [x] New feature / bug fix has been committed following the [Contribution guide](https://github.com/Qbeast-io/qbeast-spark/blob/main/CONTRIBUTING.md).
- [x] Add comments to the code (make it easier for the community!).
- [ ] Change the documentation.
- [x] Add tests.
- [ ] Your branch is updated to the main branch (dependent changes have been merged).

## How Has This Been Tested? (Optional)
- [x] It works with the parquet format, partitioned or not
- [x] It works with the delta format, partitioned or not
- [x] It extracts record count from the log if it exists; otherwise, it's retrieved from the file metadata
- [x] The operation is idempotent
- [x] It's able to infer the table format; an error is thrown if the format is not a supported one
- [x] It should not break Analyze and Optimize
- [x] It should not break Compaction
- [ ] Intelligent min max retrieval for indexing columns from AddFile stats if exists

I've created `ConvertToQbeastTest` under `utils` to make sure the solution works properly.

### CODE REVIEW

**Comment by osopardo1** [comment]
I close it for now, I would reopen it once we finish #98

**Comment by osopardo1** [comment]
Related issues to this Operation are documented on #103 and #102.

There are two ways in which we process Files without Qbeast Metadata:
1. Read. When we read the `DeltaLog` and build an `IndexStatus`, we can process those files without **qbeast tags** separately and include a fake `weight`, `cube`, `elementCount`...
2. Write. While ConvertingToQbeast, we can rewrite the Log directly with the extra metadata.

File: src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala [hunk 1]
```diff
@@ -55,7 +55,9 @@ private[delta] class IndexStatusBuilder(
 
     import spark.implicits._
     val ndims: Int = rev.transformations.size
-    revisionFiles
+    // TODO some files may not include metadata
+    val filesWithMetadata = revisionFiles.where("tags is not null")
```
Comment (Jiaweihu08) [inline src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala:59]: With the changes made, is this still necessary?

Comment (osopardo1) [reply src/main/scala/io/qbeast/spark/delta/IndexStatusBuilder.scala:59]: I don't know if we should support the reading of files without Qbeast Metadata. This case would only happen if the user does compaction with Delta (or other underlying format). 
We can solve this in another PR.

**Comment by Jiaweihu08** [comment]
> Hey @Jiaweihu08 ! Since we are prioritizing other features (such as #141 or #133 ), I think it's best to close this PR for now and get back at any time. Is it ok for you?

Agree

**Pull request closed without merging**